microk8s-resources/default-hooks/remove.d/10-cni-link

```diff
@@ -4,11 +4,15 @@
 
 if ! is_strict || (is_strict && snapctl is-connected network-control)
 then
-  for link in cni0 calico.vxlan
+  for link in cni0
   do
     if "${SNAP}/sbin/ip" link show "${link}"
     then
       "${SNAP}/sbin/ip" link delete "${link}" || true
     fi
   done
+  for calink in $("${SNAP}/sbin/ip" -j --pretty link show |\
+                 awk '/cali/ {print $2}' | tr -d '",')
```

Comment:
```suggestion
  for calink in $("${SNAP}/sbin/ip" -j link show | "${SNAP}/usr/bin/jq" '.[].ifname' -r | grep -e '\(vxlan.calico\|cali[a-f0-9]*$\)'
```

Reply:
Hey, this is great feedback, thanks a lot! I wasn't even aware that `jq` is present in the snap - I agree it's better to use it here. What do you think about this:

```shell
for calink in $("${SNAP}/sbin/ip" -j link show | "${SNAP}/usr/bin/jq" -r '.[].ifname | select(test("cali[a-f0-9]"))'
do
...
```

This will extract all calico interfaces and rely on jq only, it is a bit shorter and hopefully as clear as the grep snippet.

I noticed that the snap `ip` does not have a `--pretty` output when I built the snap locally and tested it, the version of `ip` I'm using locally renders JSON in one line without the `--pretty` flag, but with using jq, it won't matter.

And finally, thanks a lot for all the great feedback and the time spent on this PR, I've learned a lot and the discussion is great!